command/connect/envoy/bootstrap_config.go

```diff
@@ -416,11 +430,11 @@ func (c *BootstrapConfig) generatePrometheusConfig(args *BootstrapTplArgs) error
 										"routes": [
 											{
 												"match": {
-													"path": "/metrics"
+													"path": "` + path + `"
```

Comment:
is this sufficient to proxy the whole prefix? I think it makes sense while we are here to expose anything under `/stats` in this mode so you can use it for Prometheus as well. There are a few other debugging end points like `/stats/recentlookups` and maybe more will be added in the future so exposing all of it here seems most useful.

I think this would need to become `prefix` per https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto.html?highlight=route%20match#envoy-v3-api-msg-config-route-v3-routematch

What do you think?

Reply:
You're absolutely right. You had expressed it clearly, but it flew over my head. I've changed this, and added a testcase as well, checking that `/stats/prometheus` is being exposed.